### ISSUE TITLE
Easy way to extract field for plotting?

### ISSUE THREAD

**Issue by aderzelle** [knausb/vcfR#151]
Hello,

let's say I have a vcf and I want to plot the FORMAT VAF vs the FORMAT GQ 

```
#CHROM	POS	ID	REF	ALT	QUAL	FILTER	INFO	FORMAT	default
chromosome_1	6340194	.	G	C,<*>	33.3	PASS	.	GT:GQ:DP:AD:VAF:PL	0/1:33:301:164,132,0:0.438538,0:33,0,65,990,990,990
```
Is there an easy way to deal with the fact the VAF field is 0.438538**,**0?
Obviously, this doesn't work:

```
vcf = read.vcfR("ARCcestor.g.vcf.gz", verbose = FALSE)
ext = vcfR2tidy(vcf, single_frame = TRUE, info_types = TRUE, format_types = TRUE)
DT = ext$dat
DT$gt_VAF=as.numeric(as.factor(DT$gt_VAF))
```
As the presence of the comma produces completely wrong numeric values. 
I realise this might not be a question specific to vcfR but maybe of general R coding but maybe I am missing something?  

Thanks for any help.

**Comment by knausb** [comment]
How about this.

``` r
library(vcfR)
#> 
#>    *****       ***   vcfR   ***       *****
#>    This is vcfR 1.9.0.9000 
#>      browseVignettes('vcfR') # Documentation
#>      citation('vcfR') # Citation
#>    *****       *****      *****       *****
data("vcfR_test")

# Fabricate an example
# First the @fix slot
vcfR_test@fix[1, ] <- c("chromosome_1", "6340194", ".", "G", "C,<*>",   "33.3", "PASS", ".")
vcfR_test@fix[2, ] <- c("chromosome_1", "6340194", ".", "G", "C,<*>",   "33.3", "PASS", ".")
vcfR_test@fix[2,2] <- "6340204"
vcfR_test@fix <- vcfR_test@fix[-c(3:5),]
is.na(vcfR_test@fix[,"ID"]) <- TRUE

# And the @gt slot
vcfR_test@gt <- vcfR_test@gt[, 1:2]
vcfR_test@gt[1, ] <- c("GT:GQ:DP:AD:VAF:PL", "0/1:33:301:164,132,0:0.438538,0:33,0,65,990,990,990")
vcfR_test@gt[2, ] <- c("GT:GQ:DP:AD:VAF:PL", "0/1:33:301:164,132,0:0.438538,0:33,0,65,990,990,990")
vcfR_test@gt <- vcfR_test@gt[-c(3:5),]

# Extract VAF
vaf <- extract.gt(vcfR_test, element = "VAF")
allele1 <- masplit(vaf, record = 1)
allele2 <- masplit(vaf, record = 2)

# Extract GQ
gq <- extract.gt(vcfR_test, element = "GQ", as.numeric = TRUE)

# Plot
plot(gq[,1], allele1[,1])
```

![](https://i.imgur.com/4jUVYyO.png)

<sup>Created on 2020-02-04 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

I don't know what VAF is, so just trying stuff I've used elsewhere. In your VCF example data VAF is numerics delimited by a comma, which my solution should work for. In your text you've added non-numerics, so I'm not sure if that needs to be addressed.

Hope that helps!
Brian

**Comment by aderzelle** [comment]
Thanks for the answer, in the meanwhile I came out with this solution 

```
vcf = read.vcfR("ARCcestor.g.vcf.gz", verbose = FALSE)
ext = vcfR2tidy(vcf, single_frame = TRUE, info_types = TRUE, format_types = TRUE)
DT = ext$dat
DT = filter(DT, !is.na(gt_VAF))
DT$gt_VAF = sub(",.*", "", DT$gt_VAF) #getting rid of the comma and anything following number in the vaf field
DT$gt_VAF=as.numeric(as.character(DT$gt_VAF)) #converting to numeric
```
The hardest part was to remove the comma and what followed and then convert to numeric. 

Now it works great with ggplot2

```
pal3 <- wes_palette("Zissou1", 21, type = "continuous")
p = DT %>% filter(!is.na(gt_VAF)) %>% ggplot(aes(x = gt_GQ, y = gt_VAF)) + geom_hex(bins = 10) +  scale_fill_gradientn(colours = pal3)
p = p + xlab("GQ") + ylab("VAF") + facet_wrap(~CHROM) 
p
```
